### PR TITLE
initial commit for recent-trending tag module

### PR DESCRIPTION
**Pull request by ChaltinPagan**

### CODE REVIEW

File: src/components/modules/RecentAndTrending/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import { ModuleBox, OverlapScaffold, Post, Pattern } from '@/components/modules';
+import { Container, Row, Image } from '@/components/primitives';
+import { Responsive } from '@/components/utils';
+import { BREAKPOINTS, CONTAINER_PADDING, COLOR_MAP } from '@/constants';
+import { TagTileBoxModuleData, PostType, OFFSET_DIRECTION, GridOrder } from '@/types';
+import PostGrid  from './PostGrid';
+import TrendingLine from '@/assets/images/trending-line.svg';
+
+interface Props extends TagTileBoxModuleData {
+    postsOne: PostType[];
```
Comment (kitwestneat) [inline src/components/modules/RecentAndTrending/index.tsx]: Let's call these recentPosts and trendingPosts just to make it clear

File: src/components/modules/RecentAndTrending/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            {({ minWidth }) => {
+                const isDesktop = minWidth(BREAKPOINTS.LG);
+
+                const localFontStyles = isDesktop
+                    ? { fontSize: 24, lineHeight: 29 }
+                    : { fontSize: 20, lineHeight: 24 };
+                return (
+                    <Container type='content' style={{flexDirection: isDesktop? 'row' : 'column', justifyContent: 'space-between'}}>
+                      <View style={{width: isDesktop ? '68%' : '100%'}}>
+                        <Text style={[styles.headerText, localFontStyles, {marginLeft: isDesktop ? 0 : '15px'}]}>Recently in {capitalize(tag)}</Text>
```
Comment (kitwestneat) [inline src/components/modules/RecentAndTrending/index.tsx]: This view could be encapsulated as a Recent component

File: src/components/modules/RecentAndTrending/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                                    ) : (
+                                        <PostGrid order={2} itemsPerRow={1} posts={secondaryPosts} />
+                                    )}
+                                    </ModuleBox>
+                                </OverlapScaffold.Overlap>
+                            </OverlapScaffold>
+                        </Row>
+                      </View>
+
+                      <View style={[isDesktop ? styles.trendingDesktopContainer : styles.trendingMobileContainer]}>
```
Comment (kitwestneat) [inline src/components/modules/RecentAndTrending/index.tsx]: ...and this View could be the Trending component. Looking at the isDesktop check, I'd use the renderMobile() & renderDesktop pattern for it.

File: src/components/modules/RecentAndTrending/index.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                  );
+                }
+            }
+            </Responsive>
+        );
+    }
+}
+
+// Capitalize first letter of tag name
+const capitalize = (str: string) => {
```
Comment (kitwestneat) [inline src/components/modules/RecentAndTrending/index.tsx]: might as well move this function to the utils module

File: src/components/modules/RecentAndTrending/index.tsx (hunk 1 above)
Comment (ChaltinPagan) [reply src/components/modules/RecentAndTrending/index.tsx]: Renamed these props.

File: src/components/modules/RecentAndTrending/index.tsx (hunk 4 above)
Comment (ChaltinPagan) [reply src/components/modules/RecentAndTrending/index.tsx]: Moved to utils

File: src/components/modules/RecentAndTrending/index.tsx (hunk 2 above)
Comment (ChaltinPagan) [reply src/components/modules/RecentAndTrending/index.tsx]: Moved to its own React component

File: src/components/modules/RecentAndTrending/index.tsx (hunk 3 above)
Comment (ChaltinPagan) [reply src/components/modules/RecentAndTrending/index.tsx]: Moved to its own React component

**Pull request merged**